includes/admin/meta-boxes/class-wc-meta-box-coupon-data.php

```diff
@@ -341,51 +343,54 @@ public static function output( $post ) {
 			</div>
 			<?php do_action( 'woocommerce_coupon_data_panels', $coupon->get_id(), $coupon ); ?>
 			<div class="clear"></div>
-		</div>
 		<?php
 	}
 
 	/**
 	 * Save meta box data.
 	 *
-	 * @param int     $post_id
-	 * @param WP_Post $post
+	 * @param int     $post_id The post ID.
+	 * @param WP_Post $post The post object.
 	 */
 	public static function save( $post_id, $post ) {
-		// Check for dupe coupons.
-		$coupon_code  = wc_format_coupon_code( $post->post_title );
-		$id_from_code = wc_get_coupon_id_by_code( $coupon_code, $post_id );
-
-		if ( $id_from_code ) {
-			WC_Admin_Meta_Boxes::add_error( __( 'Coupon code already exists - customers will use the latest coupon with this code.', 'woocommerce' ) );
+		// Not allowed, return regular value without updating meta.
+		if ( wp_verify_nonce( wp_unslash( $_POST['woocommerce_meta_nonce'] ), 'woocommerce_save_data' ) ) { // @codingStandardsIgnoreLine.
```

Comment:
We already check for nonces before hit this method, so no need to include it again.